### PR TITLE
Use id to identify persist root node

### PR DESCRIPTION
**Pull request by pandamicro**
> Makes sure these boxes are checked before submitting your PR - thank you!
> - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
> - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
> - [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
> - [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.

Re: cocos-creator/fireball#2295

Changes proposed in this pull request:
- 使用 id 来索引 Persistent Root Node

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/CCGame.js [hunk 1]
```diff
@@ -339,10 +339,12 @@ var game = /** @lends cc.game# */{
      * @param {Node} node - The node to be made persistent
      */
     addPersistRootNode: function (node) {
-        if (!(node instanceof cc.Node))
+        if (!(node instanceof cc.Node) || !node._id) {
```
Comment (jareguo) [inline cocos2d/core/CCGame.js]: 我觉得这里访问公有接口 uuid 就行了吧

File: jsb/jsb-game.js [hunk 2]
```diff
@@ -175,11 +177,11 @@ cc.js.mixin(cc.game, {
      */
     removePersistRootNode: function (node) {
         if (node !== this._ignoreRemovePersistNode) {
-            var index = this._persistRootNodes.indexOf(node);
-            if (index !== -1) {
-                this._persistRootNodes.splice(index, 1);
+            var id = node._id || '';
+            if (node === this._persistRootNodes[id]) {
+                this._persistRootNodes[id] = null;
```
Comment (jareguo) [inline jsb/jsb-game.js]: 我觉得这里用 delete 会明确一些

File: cocos2d/core/CCDirector.js [hunk 3]
```diff
@@ -481,7 +481,7 @@ cc.Director = Class.extend(/** @lends cc.Director# */{
         }
 
         // detach persist nodes
-        for (i = persistNodes.length - 1; i >= 0; --i) {
+        for (i in persistNodes) {
```
Comment (jareguo) [inline cocos2d/core/CCDirector.js]: i 能不能换个名字…… 不然以为你在遍历数组

File: cocos2d/core/CCDirector.js [hunk 4]
```diff
@@ -510,10 +510,18 @@ cc.Director = Class.extend(/** @lends cc.Director# */{
...
             sgScene = scene._sgNode;
 
-            // Re-attach persist nodes
-            for (i = 0; i < persistNodes.length; ++i) {
-                node = persistNodes[i];
-                node.parent = scene;
+            // Re-attach or replace persist nodes
+            for (id in persistNodes) {
+                node = persistNodes[id];
+                var existNode = scene.getChildByName(node.name);
```
Comment (jareguo) [inline cocos2d/core/CCDirector.js:516]: 这里应该使用 uuid 去遍历 scene 的 children 才对吧？否则有重名怎么办

Comment (pandamicro) [inline cocos2d/core/CCDirector.js:516]: 卧槽，我看看吧

Comment (jareguo) [inline cocos2d/core/CCDirector.js:516]: 不需要的话…… 那加 uuid 意义何在？

**Pull request merged**